### ISSUE TITLE
Support code only nugets

### ISSUE THREAD

**Issue by andreasohlund** [FubuMvcArchive/ripple#196]
We have a few cody only nugets so it would be nice if ripple could mimic the nuget behaviour and add the content files to the .csproj

**Comment by andreasohlund** [comment]
This is the structure I'm thinking of

http://nikcodes.com/2013/10/23/packaging-source-code-with-nuget/

@jmarnold @jeremydmiller  how hard would it be to use your csproj wrapper thingy to add remove the /App-Packages/{NameOfPackage}.{SemVer} folder on each restore?

**Comment by jmarnold** [comment]
It wouldn't be hard to emit that. However...I'm pretty strongly opposed to embedding the version number into a folder for the same reasons we stripped it from the packages directories.

That raises a few questions:
- When do we update the file?
- Do we add a new flag to the restore command to allow you to force override the code-only nugets?
- Do we only override when you explicitly update?
- Always restore when you can't find it? (In the event you add it to .gitignore)
- Always restore (no room for local modifications)?

I'm going to have to defer to your team on this one. We don't use code-only nugets so I'd be guessing too much.

**Comment by jmarnold** [comment]
Also, are you wanting to support the file transforms (e.g., $rootnamespace$)? I think it would be possible but our implementation would definitely end up being a subset of what they do.

**Comment by andreasohlund** [comment]
Agree on skipping the version in the folder name. I don't see a need for transform at this stage

**Comment by jeremydmiller** [comment]
@jmarnold,

If it came down to it, I think we could leverage the templating support in FubuCsProjFile to apply at least the basic transforms.

**Comment by andreasohlund** [comment]
So I guess what I need is to "sync" a folder with the csproj i
1. Pull and extract nuget contents dir to: App_Package/{NameOfNuget}
2. Go through the csproj and remove/add files according to the contents of
   the folder

Any pointers where I can get started?

**Comment by jmarnold** [comment]
I'd do it on the 3.0 branch since that'll minimize your conflicts. I would do it as a ripple step (probably after the ExplodeDownloadedNugets). This call right here returns IPackage (from Nuget.Core):

https://github.com/DarthFubuMVC/ripple/blob/3.0/src/ripple/Steps/ExplodeDownloadedNugets.cs#L26

You might save those and push them into the runner for the next step to use them.

In that next step, you would have access to the solution, projects, and IPackage objects. The Project objects on the solution expose the project file from FubuCsProj so you could do anything you needed there.

**Comment by jmarnold** [comment]
@jeremydmiller 

Yeah, I was thinking that for each project you could read in the RootNamespace and at least push that through the templating. Though, I'm happy to leave that alone if we don't need it just yet.

**Comment by andreasohlund** [comment]
Cool, I'll see if I can get it going
